numpy/core/setup.py

```diff
@@ -941,7 +931,7 @@ def generate_umath_c(ext, build_dir):
                                  ],
                          depends=deps + multiarray_deps + umath_deps +
                                 common_deps,
-                         libraries=['npymath', 'npysort'],
+                         libraries=['npymath'],
```

Comment:
#17127 shows that `npymath` is also not needed here, and in fact contains symbols that are already in the other object files which leads to build failures with strict linker settings.

Reply:
I don't know that we can just get rid of `npymath`, though, it gets installed to numpy/core/lib/ for other projects to link against.  I'm not sure how common that is, but at least scipy is using it.

Reply:
This doesn't stop npymath from being built, it just stops it being passed to the linker since it does not contain any symbols that are not already available through the object files created from the compiled C files (and in fact contains many identically named functions).